### PR TITLE
:mortar_board: Topic20 Recursive binary search

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Add recursive binary search to content
Add small notes saying that the algo, other than being recursive, is ultimately the same idea

### CODE REVIEW

File: src/site/topic20.rst [hunk 1]
```diff
@@ -101,6 +101,30 @@ Iterative
...
+        } else if (haystack[midpoint].compareTo(needle) > 0) {
+            highIndex = midpoint - 1;
+        } else {
+            lowIndex = midpoint + 1;
+        }
+        return recursiveBinarySearch(needle, haystack, lowIndex, highIndex);
+    }
+
+* If I wanted to call this method, I would start with ``lowIndex`` as ``0`` and ``highIndex`` as ``someHaystack.length``
+    * ``recursiveBinarySearch(someNeedle, someHaystack, 0, someHaystack.length)``
```
Comment (twentylemon) [inline src/site/topic20.rst:126]: You'd probably define a different method with only the first two arguments, but sure. Why not.

File: src/site/topic20.rst [hunk 2]
```diff
@@ -101,6 +101,30 @@ Iterative
...
+        }
+        int midpoint = lowIndex + (highIndex - lowIndex) / 2;
+        if (haystack[midpoint].equals(needle)) {
+            return midpoint;
+        } else if (haystack[midpoint].compareTo(needle) > 0) {
+            highIndex = midpoint - 1;
+        } else {
+            lowIndex = midpoint + 1;
+        }
+        return recursiveBinarySearch(needle, haystack, lowIndex, highIndex);
```
Comment (twentylemon) [inline src/site/topic20.rst:122]: A little strange to modify the arguments. I'd rather see something like `return recursiveBinarySearch(needle, haystack, midpoint + 1, highIndex)` in the conditional blocks.

**Review by twentylemon — APPROVED** [review]

File: src/site/topic20.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/topic20.rst:126]: I'm going to make a new PR for making a point about this.

File: src/site/topic20.rst (hunk 2 above)
Comment (jameshughes89) [reply src/site/topic20.rst:122]: I will make a new PR to change this in the class and in the content.

**Pull request merged**